Sources/KituraNet/HTTP/HTTPServer.swift

```diff
@@ -145,6 +145,10 @@ public class HTTPServer: Server {
 #endif
     }
 
+    public init(eventLoopGroup: EventLoopGroup) {
```

Comment:
It's going to be difficult to modify Kitura in a non-breaking way to call this new API.

Rather than requiring the ELG to be passed in at initialization, perhaps you could make the `eventLoopGroup` property lazily initialized?  Leave `init()` with no parameters, and I think something like this will work:

```swift
// A lazily initialized EventLoopGroup, accessed via `eventLoopGroup`
private var _eventLoopGroup: EventLoopGroup?

/// The EventLoopGroup used by this HTTPServer. This property may be assigned
/// once and once only, by calling `setEventLoopGroup(value:)` before `listen()` is called.
public var eventLoopGroup: EventLoopGroup {
    if let value = self._eventLoopGroup { return value }
    #if os(Linux)
        let numberOfCores = Int(linux_sched_getaffinity())
        let value = MultiThreadedEventLoopGroup(numberOfThreads: numberOfCores > 0 ? numberOfCores : System.coreCount)
    #else
        let value = MultiThreadedEventLoopGroup(numberOfThreads: System.coreCount)
    #endif
    self._eventLoopGroup = value
    return value
}

/// Sets the EventLoopGroup to be used by this HTTPServer. This may be called once
/// and once only, and must be called prior to `listen()`.
/// - Throws: If the EventLoopGroup has already been assigned.
public func setEventLoopGroup(_ value: EventLoopGroup) throws {
    guard _eventLoopGroup == nil else {
        throw HTTPServerError.eventLoopGroupAlreadyInitialized
    }
    _eventLoopGroup = value
}
```